experimental/traffic-portal/src/app/core/cdns/cdn-detail/cdn-detail.component.html

```diff
@@ -0,0 +1,41 @@
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<mat-card>
+	<tp-loading *ngIf="!cdn"></tp-loading>
+	<form ngNativeValidate (ngSubmit)="submit($event)" *ngIf="cdn">
+		<mat-card-content>
+			<mat-form-field>
+				<mat-label>Name</mat-label>
+				<input matInput type="text" name="name" required [(ngModel)]="cdn.name">
```

Comment:
This lets me enter strings that aren't valid, and relies on the API to stop me. It's a better UX to not allow forms to be invalid. I believe the API only allows `^(\d[a-zA-Z]\.-)+$` (HTML `pattern` attributes are whole-match, so you can omit <kbd>^</kbd> and <kbd>$</kbd> whenever you're using it; it's implied).

I realize that TPv1 lets you do this too, but that's just an insufficiency of TPv1, which should be fixed if we weren't simply replacing it.

Reply:
Added a `pattern` for CDN name in 3528126e87.